eventlet/green/ssl.py

```diff
@@ -40,6 +39,63 @@ class GreenSSLSocket(_original_sslsocket):
     settimeout(), and to close/reopen the connection when a timeout
     occurs at an unexpected juncture in the code.
     """
+
+    class SwapBackOriginal:
+        SYMBOLS = ['SSLSocket', 'SSLContext']
+
+        def __init__(self):
+            self.need_swap = False
+            self.temp_objs = {}
+
+        def __enter__(self):
+            self.need_swap = 'ssl' in sys.modules
+
+            if not self.need_swap:
+                return
+            imp.acquire_lock()
+            ssl_imported = sys.modules.get('ssl')
+            for sym in GreenSSLSocket.SwapBackOriginal.SYMBOLS:
+                self.temp_objs[sym] = getattr(ssl_imported, sym, None)
+                setattr(ssl_imported, 'SSLSocket', _original_sslsocket)
+                setattr(ssl_imported, 'SSLContext', _original_sslcontext)
+
+        def __exit__(self, *args):
+            if not self.need_swap:
+                return
+            ssl_imported = sys.modules.get('ssl')
+            for sym in GreenSSLSocket.SwapBackOriginal.SYMBOLS:
+                setattr(ssl_imported, sym, self.temp_objs[sym])
+            imp.release_lock()
+
+    def __new__(cls, sock=None, keyfile=None, certfile=None,
+                server_side=False, cert_reqs=CERT_NONE,
+                ssl_version=PROTOCOL_SSLv23, ca_certs=None,
+                do_handshake_on_connect=True, *args, **kw):
+        if sys.version_info >= (3, 7):
+            if not isinstance(sock, GreenSocket):
+                sock = GreenSocket(sock)
+            with GreenSSLSocket.SwapBackOriginal():
+                ret = _original_wrap_socket(
+                    sock=sock.fd,
+                    keyfile=keyfile,
+                    certfile=certfile,
+                    server_side=server_side,
+                    cert_reqs=cert_reqs,
+                    ssl_version=ssl_version,
+                    ca_certs=ca_certs,
+                    do_handshake_on_connect=False,
+                    *args, **kw
+                )
+            ret.keyfile = keyfile
+            ret.certfile = certfile
+            ret.cert_reqs = cert_reqs
+            ret.ssl_version = ssl_version
+            ret.ca_certs = ca_certs
+            ret.__class__ = GreenSSLSocket
+            return ret
+        else:
+            return super(GreenSSLSocket, cls).__new__(cls)
```

Comment:
If one branch of an `if` / `else` is much shorter (fewer lines of code) than the other, I generally prefer to express the condition so as to put the short branch with `if` and the long branch with `else`. That way, as you read the top of each branch, the condition that brought you there is just a glance away.

Reply:
done